core/src/main/scala/chisel3/probe/package.scala

```diff
@@ -8,19 +8,23 @@ import chisel3.internal.Builder.pushCommand
 import chisel3.internal.firrtl._
 import chisel3.Data.ProbeInfo
 import chisel3.experimental.SourceInfo
-import chisel3.reflect.DataMirror.{checkTypeEquivalence, hasProbeTypeModifier}
+import chisel3.reflect.DataMirror.{checkTypeEquivalence, collectAllMembers, collectLeafMembers, hasProbeTypeModifier}
 
 import scala.language.experimental.macros
 
 package object probe extends SourceInfoDoc {
 
+  /** Set the probe information of the [[Data]] — if dealing with an
+    * [[Aggregate]], set the probe information for leaf elements only.
+    */
   private[chisel3] def setProbeModifier[T <: Data](data: T, probeInfo: Option[ProbeInfo]): Unit = {
     probeInfo.foreach { _ =>
-      data.probeInfo = probeInfo
-      data match {
-        case a: Aggregate =>
-          a.elementsIterator.foreach { e => setProbeModifier(e, probeInfo) }
-        case _ => // do nothing
+      collectAllMembers(data).foreach { e =>
+        e match {
+          case v: Vec[_]    => setProbeModifier(v.sample_element, probeInfo)
```

Comment:
i wish you could just grab the leaves... for the leaves of the vector does it do this twice?

Reply:
getting the leaves of vecs doesn't grab it's `sampleElement`, which is what the converter uses to get the element type. I'll add in a comment explaining what's going on here, but I don't know how else to work around this :(

Reply:
This also shouldn't repeat on Vec leaves, since those will all hit the last case.

The first Vec case sets probe modifiers on the sample elements of all internal Vecs.